src/main/groovy/com/gradleup/jaxb/Jaxb2Plugin.groovy

```diff
@@ -71,12 +72,58 @@ class Jaxb2Plugin implements Plugin<Project> {
     // add new tasks for creating/cleaning the auto-value sources dir
     project.task(type: CleanJaxb2SourcesDir, "cleanJaxb2SourcesDir")
     project.task(type: InitJaxb2SourcesDir, "initJaxb2SourcesDir")
-    project.task(type: GenerateJaxb2Classes, "generateJaxb2Classes")
 
     // make 'compileJava' require the new task, so that all sources are available
     project.tasks.clean.dependsOn project.tasks.cleanJaxb2SourcesDir
-    project.tasks.generateJaxb2Classes.dependsOn project.tasks.initJaxb2SourcesDir
-    project.tasks.compileJava.dependsOn project.tasks.generateJaxb2Classes
+
+    project.afterEvaluate {
+      Set<XjcTaskConfig> xjcConfigs = project.extensions.jaxb2.xjc
+      for (XjcTaskConfig xjcConfig : xjcConfigs) {
+        registerGenerateJaxb2ClassesTask(project, xjcConfig)
+      }
+    }
+  }
+
+  private static void registerGenerateJaxb2ClassesTask(Project project, XjcTaskConfig xjcConfig) {
```

Comment:
Are the `static` required?

Reply:
I don't believe it is strictly required, but the pre-existing methods that call this new method are static, so it is required in that sense. But I don't know why those pre-existing methods are static - they don't have to be.

Reply:
Makes sense 👍